### PR TITLE
Make option `level_compaction_dynamic_level_bytes` true by default

### PR DESCRIPTION
**Pull request by cbi42**
Summary: after #11321 and #11340 (both included in RocksDB v8.2), migration from `level_compaction_dynamic_level_bytes=false` to `level_compaction_dynamic_level_bytes=true` is automatic by RocksDB and requires no manual compaction from user. Making the option true by default as it has several advantages: 1. better space amplification guarantee (a more stable LSM shape). 2. compaction is more adaptive to write traffic. 3. automatic draining of unneeded levels. Wiki is updated with more detail: https://github.com/facebook/rocksdb/wiki/Leveled-Compaction#option-level_compaction_dynamic_level_bytes-and-levels-target-size. 

The PR mostly contains fixes for unit tests as they assumed `level_compaction_dynamic_level_bytes=false`. Most notable change is commit https://github.com/facebook/rocksdb/pull/11525/commits/f742be330ca1a7abc33107b00df99818f71c387b and https://github.com/facebook/rocksdb/pull/11525/commits/b1928e42b34c0d4f1cc5d5239149870c6dc7a737 which override the default option in DBTestBase to still set `level_compaction_dynamic_level_bytes=false` by default. This helps to reduce the change needed for unit tests. I think this default option override in unit tests is okay since the behavior of `level_compaction_dynamic_level_bytes=true` is tested by explicitly setting this option. Also, `level_compaction_dynamic_level_bytes=false` may be more desired in unit tests as it makes it easier to create a desired LSM shape.

Comment for option `level_compaction_dynamic_level_bytes` is updated to reflect this change and change made in #10057.

Test plan: `make -j32 J=32 check` several times to try to catch flaky tests due to this option change.

### CODE REVIEW

File: include/rocksdb/advanced_options.h [hunk 1]
```diff
@@ -642,60 +642,32 @@ struct AdvancedColumnFamilyOptions {
...
+  // More adaptive compaction to write traffic:
+  // Compaction priority will take into account estimated bytes to be compacted
+  // down to a level and favors compacting lower levels when there is a write
+  // traffic spike. Refers to
+  // https://github.com/facebook/rocksdb/wiki/Leveled-Compactio#option-level_compaction_dynamic_level_bytes-and-levels-target-size
+  // for more detailed description. See more implementation detail in:
+  // VersionStorageInfo::ComputeCompactionScore().
   //
+  // MIGRATION from level_compaction_dynamic_level_bytes=false to
+  // level_compaction_dynamic_level_bytes=true: refers to
```
Comment (ajkr) [inline include/rocksdb/advanced_options.h:658]: s/refers/refer/

File: include/rocksdb/advanced_options.h [hunk 2]
```diff
@@ -642,60 +642,32 @@ struct AdvancedColumnFamilyOptions {
...
-  // L4: f2, f3
-  // L5: f4
-  // L6: f5
-  //
   // If `allow_ingest_behind=true` or `preclude_last_level_data_seconds > 0`,
   // then the last level is reserved, and we will start filling LSM from the
-  // second last level (L5 in the above example).
+  // second last level.
+  //
+  // More adaptive compaction to write traffic:
```
Comment (ajkr) [inline include/rocksdb/advanced_options.h:649]: Can we be verbose and say "With this option on, compaction is more adaptive to write traffic"?

**Review by ajkr — APPROVED** [review]
LGTM, thanks!

File: include/rocksdb/advanced_options.h (hunk 2 above)
Comment (cbi42) [reply include/rocksdb/advanced_options.h]: Yes, updated.

**Pull request closed without merging**